File: android/app/src/main/kotlin/com/goalmaker/app/domain/update/ReleaseManifest.kt
package com.goalmaker.app.domain.update

import com.goalmaker.app.domain.version.SemanticVersion

/** The latest release as described by a verified release manifest (contracts/schemas). */
data class ReleaseManifest(
    val version: SemanticVersion,
    val androidVersionCode: Int?,
    val publishedAt: String,
    val notes: String?,
    val artifacts: List<ReleaseArtifact>,
) {
    fun artifactFor(platform: ReleasePlatform): ReleaseArtifact? = artifacts.firstOrNull { it.platform == platform }
}

File: windows/src/GoalMaker.Core/Updates/ReleaseManifest.cs
using GoalMaker.Core.Versioning;

namespace GoalMaker.Core.Updates;

/// <summary>The latest release as described by a verified release manifest (contracts/schemas).</summary>
public sealed record ReleaseManifest(
    SemanticVersion Version,
    int? AndroidVersionCode,
    string PublishedAt,
    string? Notes,
    IReadOnlyList<ReleaseArtifact> Artifacts)
{
    public ReleaseArtifact? ArtifactFor(ReleasePlatform platform) =>
        Artifacts.FirstOrDefault(artifact => artifact.Platform == platform);
}
